compiler/int128.nim

```diff
@@ -540,7 +540,26 @@ proc ldexp(x: float64, exp: cint): float64 {.importc: "ldexp", header: "<math.h>
 
 template bitor(a, b, c: Int128): Int128 = bitor(bitor(a, b), c)
 
+const lowerBoundF64 = -pow(2'f64, 127)
+# 2^127-1 can't be represented with float64 precision, therefore upper bound is
+# Exclusive
+const upperBoundF64 = +pow(2'f64, 127)
+
+proc inInt128Range*(arg: float64): bool =
+  ## Simple range check. Of coures NaN is defined not to be in range.
+  lowerBoundF64 <= arg and arg < upperBoundF64
+
 proc toInt128*(arg: float64): Int128 =
+  # actually enabling this assert will cause /tests/misc/tconv.nim to fail
+  # assert(inInt128Range(arg), "out of range")
+
+  # saturating at `Max` and `Min` values isn't correct, but better than returing
+  # complete garbage.
+  if arg < lowerBoundF64:
+    return Min
+  if upperBoundF64 <= arg:
+    return Max
+
```

Comment:
We should cover NaN, too (will need an import from `math`):

```suggestion
  if isNan(arg):
    return 0

```

The pick of `0` is motivated by [WASM](https://github.com/WebAssembly/spec/blob/main/proposals/nontrapping-float-to-int-conversion/Overview.md#design) and [Rust](https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast).

Reply:
I would have picked 0xDEADBEEF or something. But 0 is fine for me as well.